### PR TITLE
#742 CI: Add lint checks

### LINKED ISSUE #742: Add lint checks on CI
### Discussed in https://github.com/honeybadger-io/honeybadger-js/discussions/738

<div type='discussions-op-text'>

<sup>Originally posted by **subzero10** April  3, 2022</sup>
I have noticed that for some projects we have eslint enabled but it is only executed manually. For some projects, we don't have eslint configured at all.

Personally, I use my IDE to automatically lint on file save when eslint is configured on a project. However, this is not a guaranteed solution for all contributors.

I think we should add lint checks on CI for all projects. Or even a git hook to lint before commit. Or both. 

What do you think?</div>

### PR DESCRIPTION
**Pull request by subzero10**
## Status
**READY**

## Description
Fixes #742 
Enables lint checks on CI.

**Bonus**: Cache npm installations

FYI: From your IDE, you can have [eslint automatically fix the file](https://www.jetbrains.com/help/idea/eslint.html#ws_eslint_configure_run_eslint_on_save) you are working on:
image

## Todos
- [x] Review current lint config
- [x] Lint on CI

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1 @@
+**/bundle.js
```
Comment (subzero10) [inline .eslintignore:1]: Ignore bundle files.

File: .eslintrc [hunk 2]
```diff
@@ -10,7 +10,17 @@
   ],
   "parser": "@typescript-eslint/parser",
   "rules": {
+    "quotes": [
```
Comment (subzero10) [inline .eslintrc:13]: For consistency, enforce single quotes whenever possible.

File: .eslintrc [hunk 3]
```diff
@@ -10,7 +10,17 @@
...
+      "error",
+      "single"
+    ],
     "indent": ["error", 2],
+    "no-var": "error",
```
Comment (subzero10) [inline .eslintrc:18]: Enforce `const` & `let` instead of `var`.

File: .eslintrc [hunk 4]
```diff
@@ -10,7 +10,17 @@
...
+    "object-curly-spacing": [
```
Comment (subzero10) [inline .eslintrc:19]: Enforce style consistency in object declarations.

File: .eslintrc [hunk 5]
```diff
@@ -21,8 +31,7 @@
   },
   "ignorePatterns": [
     "test/integration/",
-    "dist",
-    "examples"
```
Comment (subzero10) [inline .eslintrc:25]: I enabled eslint on examples folder, hence the number of changed files on this PR.

File: .eslintrc [hunk 6]
```diff
@@ -32,6 +41,7 @@
   {
     "browser": true,
     "node": true,
-    "es6": true
+    "es6": true,
+    "webextensions": true
```
Comment (subzero10) [inline .eslintrc:45]: For the chrome extension example.

File: .github/workflows/node.js.yml [hunk 7]
```diff
@@ -8,52 +8,86 @@ on:
...
-    - run: npm run tsd
+        cache: 'npm'
+
+    - name: Build
+      run: npm ci
+
+    - name: Check Typescript type definitions
+      run: npm run tsd
+
+  lint:
```
Comment (subzero10) [inline .github/workflows/node.js.yml:77]: The lint step.

File: .github/workflows/node.js.yml [hunk 8]
```diff
@@ -8,52 +8,86 @@ on:
...
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Setup Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v3
       with:
         node-version: ${{ matrix.node-version }}
-    - run: npm ci
-    - run: npm test
+        cache: 'npm'
```
Comment (subzero10) [inline .github/workflows/node.js.yml:23]: Added caching in all setup-node steps.

**Review by shalvah — APPROVED** [review]

**Pull request merged**